scripts/cmake/git-submodules.cmake

```diff
@@ -5,7 +5,36 @@ set(RIMAGE_CMAKE "${SOF_ROOT_SOURCE_DIRECTORY}/rimage/CMakeLists.txt")
 
 if(GIT_FOUND AND EXISTS "${SOF_ROOT_SOURCE_DIRECTORY}/.git")
 
-	if(NOT EXISTS "${RIMAGE_CMAKE}")
+	# You can defeat this check by manually initializing rimage and
+	# _not_ some other submodule = only if you are actively trying
+	# to make your own life miserable.
+	if(EXISTS "${RIMAGE_CMAKE}")
```

Comment:
Could maybe we add an env var check also to override this so someone doesnt have to re-init the repo to bypass this if they already made the changes?

Reply:
I just added `I_KNOW_SUBMODULES=1  ./scripts/xtensa-build-all.sh -d -r apl `, enjoy.

Note the main purpose of this PR was to warn about newer versions of submodules, not missing submodules although it warns about that too.
BTW your comment prompted me to perform more testing which led me to find a gap in the MATCHES regex below, thx! Fixed.